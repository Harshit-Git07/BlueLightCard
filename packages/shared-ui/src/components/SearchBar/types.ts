export interface SearchProps {
  value?: string;
  showBackArrow?: boolean;
  onSearch: (searchTerm: string) => void;
  labelText?: string;
  onBackButtonClick?: () => void;
  onClear?: () => void;
  onFocus?: () => void;
  onBlur?: () => void;
  placeholderText?: string;
  experimentalSearchVariant?: string;
  errorMessage?: string;
}
