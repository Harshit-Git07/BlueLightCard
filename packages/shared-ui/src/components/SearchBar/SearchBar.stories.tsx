// components/SearchBar/SearchBar.stories.js (or .tsx)
import { Meta, StoryFn } from '@storybook/react';
import { fireEvent, userEvent, within } from '@storybook/testing-library';
import SearchBar from './index';

const componentMeta: Meta<typeof SearchBar> = {
  title: 'SearchBar',
  component: SearchBar,
  argTypes: {
    onSearch: { action: 'Search submitted' },
  },
};

const DefaultTemplate: StoryFn<typeof SearchBar> = (args) => <SearchBar {...args} />;

export const Default = DefaultTemplate.bind({});

Default.args = {
  labelText: 'Search bar',
  placeholderText: 'Search for offers or companies',
  showBackArrow: false,
};

export const Edited = DefaultTemplate.bind({});

Edited.args = {
  ...Default.args,
};
Edited.play = async ({ canvasElement }) => {
  const screen = within(canvasElement);

  const searchBar = screen.getByLabelText('Search bar');

  fireEvent.focusIn(searchBar);
  await userEvent.type(searchBar, 'Nike');
};

export const Error = DefaultTemplate.bind({});

Error.args = {
  ...Default.args,
  errorMessage: 'Enter 3 or more characters to search.',
};

export const Submitted = DefaultTemplate.bind({});

Submitted.args = {
  ...Default.args,
};
Submitted.play = async (props) => {
  if (Edited.play) {
    await Edited.play(props);
  }

  const { canvasElement } = props;
  const screen = within(canvasElement);

  const searchBar = screen.getByLabelText('Search bar');

  fireEvent.keyDown(searchBar, { key: 'Enter' });
};

export const SubmittedAndCleared = DefaultTemplate.bind({});
SubmittedAndCleared.args = {
  ...Default.args,
  clearOnSubmit: true,
};
SubmittedAndCleared.play = async (props) => {
  if (Submitted.play) {
    await Submitted.play(props);
  }
};

export const Cleared = DefaultTemplate.bind({});

Cleared.args = {
  ...Default.args,
};
Cleared.play = async (props) => {
  if (Edited.play) {
    await Edited.play(props);
  }

  const { canvasElement } = props;
  const screen = within(canvasElement);

  const clearButton = screen.getByLabelText('Clear button');

  await userEvent.click(clearButton.children[0]);
};

export const Reset = DefaultTemplate.bind({});

Reset.args = {
  ...Default.args,
};
Reset.play = async (props) => {
  if (Edited.play) {
    await Edited.play(props);
  }

  const { canvasElement } = props;
  const screen = within(canvasElement);

  const backButton = screen.getByLabelText('Back button');

  await userEvent.click(backButton.children[0]);
};

export const ExperimentalDark = DefaultTemplate.bind({});

ExperimentalDark.args = {
  ...Default.args,
  experimentalSearchVariant: 'background-variant-dark',
};

export const ExperimentalLight = DefaultTemplate.bind({});

ExperimentalLight.args = {
  ...Default.args,
  experimentalSearchVariant: 'background-variant-light',
};

export default componentMeta;
