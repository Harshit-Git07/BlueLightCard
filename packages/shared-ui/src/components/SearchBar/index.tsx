import {
  FC,
  useRef,
  useState,
  ChangeEventHandler,
  KeyboardEventHandler,
  useEffect,
  useCallback,
} from 'react';
import { SearchProps } from './types';
import { faSearch, faCircleXmark, faArrowLeft } from '@fortawesome/pro-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const SearchBar: FC<SearchProps> = ({
  onSearch,
  labelText,
  onClear,
  onBackButtonClick,
  onFocus,
  placeholderText,
  value,
  showBackArrow,
}) => {
  const initialValue = useRef<HTMLInputElement>(null);
  const [searchTerm, setSearchTerm] = useState<string>('');
  const [isFocused, setIsFocused] = useState<boolean>(false);
  const [errorMessage, setErrorMessage] = useState<string>('');

  const _showBackArrow = isFocused || showBackArrow;

  useEffect(() => {
    if (value) {
      setSearchTerm(value);
    }
  }, [value]);

  useEffect(() => {
    if (initialValue) {
      isFocused ? initialValue.current?.focus() : initialValue.current?.blur();
    }
  }, [isFocused]);

  const onSubmit: KeyboardEventHandler = (event) => {
    if (event.key !== 'Enter') {
      return;
    }

    event.preventDefault();

    if (searchTerm && searchTerm.length >= 3) {
      setErrorMessage('');
      setIsFocused(false);
      onSearch(searchTerm);
    } else {
      setErrorMessage('Enter 3 or more characters to search.');
    }
  };

  const onInputChange: ChangeEventHandler<HTMLInputElement> = (event) => {
    setSearchTerm(event.target.value);
  };

  const clearSearch = () => {
    if (onClear) {
      onClear();
    }

    setSearchTerm('');

    if (initialValue.current) {
      initialValue.current.focus();
    }
  };

  const _onBackButtonClick = useCallback(() => {
    if (onBackButtonClick) {
      onBackButtonClick();
    }

    setIsFocused(false);
    setSearchTerm('');
    setErrorMessage('');
  }, []);

  const onInputFocus = () => {
    if (onFocus) {
      onFocus();
    }

    setIsFocused(true);
  };

  const onInputBlur = () => {
    setIsFocused(false);
  };

  const leftArrow = (
    <button
      aria-label="Back button"
      className="absolute left-2 px-2 top-1/2 transform -translate-y-1/2"
      type="button"
      onClick={_onBackButtonClick}
    >
      <FontAwesomeIcon
        icon={faArrowLeft}
        size="xl"
        className="text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
        aria-hidden="true"
      />
    </button>
  );

  const searchIcon = (
    <FontAwesomeIcon
      icon={faSearch}
      size="xl"
      className="absolute left-2 px-2 top-1/2 transform -translate-y-1/2 text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
      aria-hidden="true"
    />
  );

  const clearIcon = (
    <button aria-label="Clear button" type="button">
      <FontAwesomeIcon
        icon={faCircleXmark}
        size="xl"
        onClick={clearSearch}
        className="absolute right-2 px-2 top-1/2 transform -translate-y-1/2 cursor-pointer text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
        aria-hidden="true"
      />
    </button>
  );

  return (
    <form className="px-2 z-10 w-full py-4">
      <div className="relative">
        {_showBackArrow ? leftArrow : searchIcon}
        <input
          aria-label={labelText}
          id="searchInput"
          ref={initialValue}
          onFocus={onInputFocus}
          onBlur={onInputBlur}
          onChange={onInputChange}
          value={searchTerm}
          placeholder={isFocused ? '' : placeholderText}
          type="search"
          enterKeyHint="search"
          autoComplete="off"
          onKeyDown={onSubmit}
          className="search-bar text-ellipsis pl-14 pr-12 py-3 rounded-full w-full overflow-x-hidden bg-searchBar-bg-colour-light dark:bg-searchBar-bg-colour-dark border-searchBar-outline-colour-light border text-searchBar-label-colour-light dark:text-searchBar-label-colour-dark dark:border-searchBar-outline-colour-dark focus:outline-none font-searchBar-label-font text-searchBar-label-font font-searchBar-label-font-weight tracking-searchBar-label-font leading-searchBar-label-font"
        />
        {searchTerm && clearIcon}
      </div>

      {errorMessage && <div className="text-colour-system-red-500">{errorMessage}</div>}
    </form>
  );
};

export default SearchBar;
