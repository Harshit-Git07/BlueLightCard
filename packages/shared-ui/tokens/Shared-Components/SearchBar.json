{
  "searchBar-bg-colour": {
    "light": {
      "value": "{colour.surface-bright.light}",
      "type": "color"
    },
    "dark": {
      "value": "{colour.surface-bright.dark}",
      "type": "color"
    }
  },
  "searchBar-outline-colour": {
    "light": {
      "value": "{colour.onSurface-outline.light}",
      "type": "color"
    },
    "dark": {
      "value": "{colour.onSurface-outline.dark}",
      "type": "color"
    }
  },
  "searchBar-label-font": {
    "value": "{typography.body}",
    "type": "typography"
  },
  "searchBar-border-radius": {
    "value": "{border-radius.round}",
    "type": "borderRadius"
  },
  "searchBar-label-colour": {
    "light": {
      "value": "{colour.onSurface-subtle.light}",
      "type": "color"
    },
    "dark": {
      "value": "{colour.onSurface-subtle.dark}",
      "type": "color"
    }
  },
  "searchBar-icon-colour": {
    "light": {
      "value": "{colour.onSurface-subtle.light}",
      "type": "color"
    },
    "dark": {
      "value": "{colour.onSurface-subtle.dark}",
      "type": "color"
    }
  }
}
