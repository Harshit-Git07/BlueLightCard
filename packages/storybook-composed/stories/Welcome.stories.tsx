import type { Meta, StoryObj } from '@storybook/react';
import WelcomeComponent from './Welcome';

const meta: Meta<typeof WelcomeComponent> = {
  component: WelcomeComponent,
};

export default meta;

type Story = StoryObj<typeof WelcomeComponent>;

export const Welcome: Story = {
  args: {},
  argTypes: {},
};
