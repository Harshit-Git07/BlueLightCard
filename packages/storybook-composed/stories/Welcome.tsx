import React, { FC } from 'react';

const WelcomeComponent: FC = () => (
  <div>
    <p>
      Welcome to the Storybook Composed instance. Here you can find all of our stories across Web,
      Mobile Hybrid, and Shared UI for every brand. Please view these stories from the sidebar.
    </p>
  </div>
);

export default WelcomeComponent;
