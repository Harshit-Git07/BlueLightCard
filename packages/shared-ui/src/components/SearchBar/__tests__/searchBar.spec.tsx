import { act, render, screen, within } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../SearchBar.stories';

const { Default, Edited, Submitted, Cleared, Reset, Error, ExperimentalDark, ExperimentalLight } =
  composeStories(stories);

it('renders the search bar', () => {
  const { container } = render(<Default />);

  expect(container).toMatchSnapshot();
});

it('renders the search bar with active experiments `dark`', async () => {
  const { container } = render(<ExperimentalDark />);

  const forms = container.getElementsByTagName('form');
  const searchForm = forms[0];
  const searchInput = await within(searchForm).findByLabelText('Search bar');

  expect(searchForm).toHaveClass('bg-colour-primary-light');
  expect(searchForm).toHaveClass('shadow-[0_4px_4px_4px_rgba(42,42,42,0.16)]');
  expect(searchInput).toHaveClass('border-searchBar-outline-colour-light');
  expect(container).toMatchSnapshot();
});

it('renders the search bar with active experiments `light`', async () => {
  const { container } = render(<ExperimentalLight />);

  const forms = container.getElementsByTagName('form');
  const searchForm = forms[0];
  const searchInput = await within(searchForm).findByLabelText('Search bar');

  expect(searchForm).toHaveClass('bg-[#F6F9FF]');
  expect(searchForm).toHaveClass('shadow-[0_4px_4px_4px_rgba(42,42,42,0.16)]');
  expect(searchInput).toHaveClass('border-colour-primary-light');
  expect(container).toMatchSnapshot();
});

it('edits the search bar', async () => {
  const onFocus = jest.fn();
  const { container } = render(<Edited onFocus={onFocus} />);

  await act(() => Edited.play({ canvasElement: container }));

  expect(onFocus).toHaveBeenCalled();
  expect(container).toMatchSnapshot();
});

it('submits the search bar', async () => {
  const onSearch = jest.fn();

  const { container } = render(<Submitted onSearch={onSearch} />);

  await act(() => Submitted.play({ canvasElement: container }));

  expect(onSearch).toHaveBeenCalledWith('Nike');
  expect(container).toMatchSnapshot();
});

it('clears the search bar', async () => {
  const onClear = jest.fn();

  const { container } = render(<Cleared onClear={onClear} />);

  await act(() => Cleared.play({ canvasElement: container }));

  expect(onClear).toHaveBeenCalled();
  expect(container).toMatchSnapshot();
});

it('resets the search bar', async () => {
  const onBackButtonClick = jest.fn();

  const { container } = render(<Reset onBackButtonClick={onBackButtonClick} />);

  await act(() => Reset.play({ canvasElement: container }));

  expect(onBackButtonClick).toHaveBeenCalled();
  expect(container).toMatchSnapshot();
});

describe('displays an error', () => {
  jest.restoreAllMocks();

  it('shows the given error message', async () => {
    const onSearch = jest.fn();

    const { container } = render(<Error onSearch={onSearch} />);

    const errorMessage = screen.getByText('Enter 3 or more characters to search.');
    expect(errorMessage).toHaveClass('error-message');

    expect(container).toMatchSnapshot();
  });
});
