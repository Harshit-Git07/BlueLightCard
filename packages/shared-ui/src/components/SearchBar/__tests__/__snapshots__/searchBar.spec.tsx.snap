// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clears the search bar 1`] = `
<div>
  <form
    class="px-2 z-10 w-full py-4"
  >
    <div
      class="relative"
    >
      <button
        aria-label="Back button"
        class="absolute left-2 px-2 top-1/2 transform -translate-y-1/2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-left fa-xl text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
          data-icon="arrow-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            fill="currentColor"
          />
        </svg>
      </button>
      <input
        aria-label="Search bar"
        autocomplete="off"
        class="search-bar text-ellipsis pl-14 pr-12 py-3 rounded-full w-full overflow-x-hidden bg-searchBar-bg-colour-light dark:bg-searchBar-bg-colour-dark border-searchBar-outline-colour-light border text-searchBar-label-colour-light dark:text-searchBar-label-colour-dark dark:border-searchBar-outline-colour-dark focus:outline-none font-searchBar-label-font text-searchBar-label-font font-searchBar-label-font-weight tracking-searchBar-label-font leading-searchBar-label-font"
        enterkeyhint="search"
        id="searchInput"
        placeholder=""
        type="search"
        value=""
      />
    </div>
  </form>
</div>
`;

exports[`edits the search bar 1`] = `
<div>
  <form
    class="px-2 z-10 w-full py-4"
  >
    <div
      class="relative"
    >
      <button
        aria-label="Back button"
        class="absolute left-2 px-2 top-1/2 transform -translate-y-1/2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-left fa-xl text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
          data-icon="arrow-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            fill="currentColor"
          />
        </svg>
      </button>
      <input
        aria-label="Search bar"
        autocomplete="off"
        class="search-bar text-ellipsis pl-14 pr-12 py-3 rounded-full w-full overflow-x-hidden bg-searchBar-bg-colour-light dark:bg-searchBar-bg-colour-dark border-searchBar-outline-colour-light border text-searchBar-label-colour-light dark:text-searchBar-label-colour-dark dark:border-searchBar-outline-colour-dark focus:outline-none font-searchBar-label-font text-searchBar-label-font font-searchBar-label-font-weight tracking-searchBar-label-font leading-searchBar-label-font"
        enterkeyhint="search"
        id="searchInput"
        placeholder=""
        type="search"
        value="Nike"
      />
      <button
        aria-label="Clear button"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-xmark fa-xl absolute right-2 px-2 top-1/2 transform -translate-y-1/2 cursor-pointer text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
          data-icon="circle-xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </form>
</div>
`;

exports[`renders the search bar 1`] = `
<div>
  <form
    class="px-2 z-10 w-full py-4"
  >
    <div
      class="relative"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-magnifying-glass fa-xl absolute left-2 px-2 top-1/2 transform -translate-y-1/2 text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
        />
      </svg>
      <input
        aria-label="Search bar"
        autocomplete="off"
        class="search-bar text-ellipsis pl-14 pr-12 py-3 rounded-full w-full overflow-x-hidden bg-searchBar-bg-colour-light dark:bg-searchBar-bg-colour-dark border-searchBar-outline-colour-light border text-searchBar-label-colour-light dark:text-searchBar-label-colour-dark dark:border-searchBar-outline-colour-dark focus:outline-none font-searchBar-label-font text-searchBar-label-font font-searchBar-label-font-weight tracking-searchBar-label-font leading-searchBar-label-font"
        enterkeyhint="search"
        id="searchInput"
        placeholder="Search for offers or companies"
        type="search"
        value=""
      />
    </div>
  </form>
</div>
`;

exports[`resets the search bar 1`] = `
<div>
  <form
    class="px-2 z-10 w-full py-4"
  >
    <div
      class="relative"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-magnifying-glass fa-xl absolute left-2 px-2 top-1/2 transform -translate-y-1/2 text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
        />
      </svg>
      <input
        aria-label="Search bar"
        autocomplete="off"
        class="search-bar text-ellipsis pl-14 pr-12 py-3 rounded-full w-full overflow-x-hidden bg-searchBar-bg-colour-light dark:bg-searchBar-bg-colour-dark border-searchBar-outline-colour-light border text-searchBar-label-colour-light dark:text-searchBar-label-colour-dark dark:border-searchBar-outline-colour-dark focus:outline-none font-searchBar-label-font text-searchBar-label-font font-searchBar-label-font-weight tracking-searchBar-label-font leading-searchBar-label-font"
        enterkeyhint="search"
        id="searchInput"
        placeholder="Search for offers or companies"
        type="search"
        value=""
      />
    </div>
  </form>
</div>
`;

exports[`submits the search bar 1`] = `
<div>
  <form
    class="px-2 z-10 w-full py-4"
  >
    <div
      class="relative"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-magnifying-glass fa-xl absolute left-2 px-2 top-1/2 transform -translate-y-1/2 text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
        />
      </svg>
      <input
        aria-label="Search bar"
        autocomplete="off"
        class="search-bar text-ellipsis pl-14 pr-12 py-3 rounded-full w-full overflow-x-hidden bg-searchBar-bg-colour-light dark:bg-searchBar-bg-colour-dark border-searchBar-outline-colour-light border text-searchBar-label-colour-light dark:text-searchBar-label-colour-dark dark:border-searchBar-outline-colour-dark focus:outline-none font-searchBar-label-font text-searchBar-label-font font-searchBar-label-font-weight tracking-searchBar-label-font leading-searchBar-label-font"
        enterkeyhint="search"
        id="searchInput"
        placeholder="Search for offers or companies"
        type="search"
        value="Nike"
      />
      <button
        aria-label="Clear button"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-xmark fa-xl absolute right-2 px-2 top-1/2 transform -translate-y-1/2 cursor-pointer text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
          data-icon="circle-xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </form>
</div>
`;
